pylate/evaluation/pylate_information_retrieval_evaluator.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import annotations
+
+import heapq
+import logging
+from contextlib import nullcontext
+from typing import TYPE_CHECKING
+
+import torch
+from sentence_transformers.evaluation import InformationRetrievalEvaluator
+from torch import Tensor
+from tqdm import trange
+
+if TYPE_CHECKING:
+    from sentence_transformers.SentenceTransformer import SentenceTransformer
+
+logger = logging.getLogger(__name__)
+
+
+class PyLateInformationRetrievalEvaluator(InformationRetrievalEvaluator):
+    """
+    This class evaluates an Information Retrieval (IR) setting.
+
+    Given a set of queries and a large corpus set. It will retrieve for each query the top-k most similar document. It measures
+    Mean Reciprocal Rank (MRR), Recall@k, and Normalized Discounted Cumulative Gain (NDCG)
+
+    Example:

```

Comment:
The docstring format won't be parsed by our doc and it could be cool to create some unit tests in this docstring